pipelines/batch/src/main/java/com/google/fhir/analytics/ParquetMerger.java

```diff
@@ -56,13 +71,74 @@ public class ParquetMerger {
 
   private static final Logger log = LoggerFactory.getLogger(ParquetMerger.class);
 
+  private static final TupleTag<GenericRecord> oldDwh = new TupleTag<>();
+
+  private static final TupleTag<GenericRecord> newDwh = new TupleTag<>();
+
   private static String ID_KEY = "id";
   private static String META_KEY = "meta";
   private static String LAST_UPDATED_KEY = "lastUpdated";
   private static String TAG_KEY = "tag";
   private static String SYSTEM_KEY = "system";
   private static String CODE_KEY = "code";
 
+  /**
+   * This method reads all the Parquet files under the paths {@code dwhFilesList} for the given
+   * {@code viewName}. It then groups the records by ID key, tags the records according to their
+   * creation timestamp and returns the grouped PCollection.
+   */
+  private static PCollection<KV<String, CoGbkResult>> readViewAndGroupById(
+      Pipeline pipeline, DwhFiles dwh1, DwhFiles dwh2, String viewName, Schema schema)
+      throws IOException {
+
+    //TODO:The extra reshuffle may be worth it to keep the logic simple and due to the smaller view
+    // schemas. Ask Chandra is the performance hit (memory-wise) won't be as large
+
+    // Reading all parquet files at once instead of one set at a time, reduces the number of Flink
+    // reshuffle operations by one.
+    PCollection<ReadableFile> inputFiles1 =
+        pipeline
+            .apply(Create.of(getParquetFilePaths(viewName, Arrays.asList(dwh1))))
+            .apply(FileIO.matchAll())
+            .apply(FileIO.readMatches());
+
+    PCollection<ReadableFile> inputFiles2 =
+        pipeline
+            .apply(Create.of(getParquetFilePaths(viewName, Arrays.asList(dwh2))))
+            .apply(FileIO.matchAll())
+            .apply(FileIO.readMatches());
+
+    PCollection<GenericRecord> records1 = inputFiles1.apply(ParquetIO.readFiles(schema));
+    PCollection<GenericRecord> records2 = inputFiles2.apply(ParquetIO.readFiles(schema));
+
+    Instant timestamp1 = dwh1.readTimestampFile(DwhFiles.TIMESTAMP_FILE_START);
+    Instant timestamp2 = dwh2.readTimestampFile(DwhFiles.TIMESTAMP_FILE_START);
+
+    PCollection<KV<String, GenericRecord>> dwhIdGroup1 =
+        records1
+            .apply(ParDo.of(new GroupViewIds()))
+            .setCoder(KvCoder.of(StringUtf8Coder.of(), AvroCoder.of(schema)));
```

Comment:
Question: Is setting this code required?

Reply:
You're right, removed.